dotnet/src/Planners/Planners.Core/Handlebars/HandlebarsTemplateEngineExtensions.cs

```diff
@@ -0,0 +1,405 @@
+{
+    /// <summary>
+    /// The key used to store the reserved output type in the dictionary of variables passed to the Handlebars template engine.
+    /// </summary>
+    public static readonly string ReservedOutputTypeKey = "RESERVED_OUTPUT_TYPE";
+
+    /// <summary>
+    /// Renders a Handlebars template in the context of a Semantic Kernel.
+    /// </summary>
+    /// <param name="kernel">The Semantic Kernel.</param>
+    /// <param name="executionContext">The execution context.</param>
+    /// <param name="template">The Handlebars template to render.</param>
+    /// <param name="variables">The dictionary of variables to pass to the Handlebars template engine.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The rendered Handlebars template.</returns>
+    public static string Render
+    (
+        IKernel kernel,
+        SKContext executionContext,
+        string template,
+        Dictionary<string, object?> variables,
+        CancellationToken cancellationToken = default
+    )
+    {
+        IHandlebars handlebarsInstance = HandlebarsDotNet.Handlebars.Create(
+            new HandlebarsConfiguration
+            {
+                NoEscape = true
+            });
+
+        // Add helpers for each function
+        foreach (FunctionView function in kernel.Functions.GetFunctionViews())
+        {
+            RegisterFunctionAsHelper(kernel, executionContext, handlebarsInstance, function, variables, cancellationToken);
+        }
+
+        // Add system helpers
+        RegisterSystemHelpers(handlebarsInstance, variables);
+
+        var compiledTemplate = handlebarsInstance.Compile(template);
+        return compiledTemplate(variables);
+    }
+
+    private static void RegisterFunctionAsHelper(
+        IKernel kernel,
+        SKContext executionContext,
+        IHandlebars handlebarsInstance,
+        FunctionView functionView,
+        Dictionary<string, object?> variables,
+        CancellationToken cancellationToken = default)
+    {
+        string fullyResolvedFunctionName = functionView.PluginName + "-" + functionView.Name;
+
+        handlebarsInstance.RegisterHelper(fullyResolvedFunctionName, (in HelperOptions options, in Context context, in Arguments arguments) =>
+        {
+            // Get the parameters from the template arguments
+            if (arguments.Any())
+            {
+                if (arguments[0].GetType() == typeof(HashParameterDictionary))
+                {
+                    // Process hash arguments
+                    var handlebarArgs = arguments[0] as IDictionary<string, object>;
+
+                    // Prepare the input parameters for the function
+                    foreach (var param in functionView.Parameters)
+                    {
+                        // TODO: accomodate ServerUrl override? 
+                        if (param.Name.Contains("server_url") || param.Name.Contains("server-url"))
```

Comment:
This can be removed after this PR: https://github.com/microsoft/semantic-kernel/pull/3427

Reply:
Awesome! Removed!